reporting/ReportingETL.cpp

```diff
@@ -331,8 +384,9 @@ ReportingETL::buildNextLedger(org::xrpl::rpc::v1::GetLedgerResponse& rawData)
 std::optional<uint32_t>
 ReportingETL::runETLPipeline(uint32_t startSequence, int numExtractors)
 {
-    if (startSequence > finishSequence_)
+    if (finishSequence_ && startSequence > *finishSequence_)
```

Comment:
As discussed, `finishSequence_` is a `std::optional`, and as such we should check if it is `std::nullopt` before doing comparision.